nativelib/src/main/scala/scala/scalanative/unsafe/package.scala

```diff
@@ -196,35 +196,71 @@ package object unsafe extends unsafe.UnsafePackageCompat {
     }
   }
 
+  private def nonNullToCString(
+      cb: CharBuffer,
+      cs: Charset,
+      charSize: CInt
+  )(implicit z: Zone): Ptr[Byte] = {
+    val buf = cs.encode(cb)
+    val off = buf.position()
+
+    val size = (buf.limit() - off).toUSize
+    val cstr = z.alloc(size + charSize.toUSize)
+    ffi.memcpy(cstr, buf.array().at(off), size)
+
+    // Set null termination bytes (z.alloc does not initialize memory)
+    (0 until charSize).foreach { cstr.update(size + _, 0.toByte) }
```

Comment:
@LeeTibbert perhaps this is less obscure (no explicit Tag.Byte and pointer manipulation)?

Reply:
Entry overlap.

Yes, 
`(0 until charSize).foreach { cstr.update(size + _, 0.toByte) }` is certainly within author's prerogative and, to
me, much easier to sight read. Very functional.

A small wonderment,  since you are dealing with a know cstring a.k.a Byte array what does the '.update' add?
Not challenging, wondering. Is '.update' an alias for '() = ' indexing with assignment  or visa versa?
I guess obscure reflects on the paths on which one has most traveled; earliest and  most recently.

`(0 until charSize).foreach { cstr(size + _) =  0.toByte) }` 

PS: Thank you for the no Tag and no extra pointer manipulation  changes.

Reply:
1. revised.
2. `cstr(_)` is read, not write, if i read the code correctly; the class defines `apply` to load, and `update` to store.